src/core/ngx_connection.c

```diff
@@ -487,8 +487,19 @@ ngx_open_listening_sockets(ngx_cycle_t *cycle)
                 continue;
             }
 
-            s = ngx_socket(ls[i].sockaddr->sa_family, ls[i].type, 0);
+            s = (ngx_socket_t) -1;
+            if (ls[i].protocol > 0) {
+                s = ngx_socket(ls[i].sockaddr->sa_family, ls[i].type,
+                               ls[i].protocol);
+                /* In case of error, retry with the default protocol */
+                ngx_log_error(NGX_LOG_NOTICE, log, 0,
```

Comment:
It looks like you will always print something, no matter if there was an error or not, no?